tests/unit/utils/format-url-test.js

```diff
@@ -0,0 +1,24 @@
+import { formatURLConditional } from 'ember-website/utils/format-url';
+import { module, test } from 'qunit';
+
+module('Unit | Utility | format-url', function () {
+  module('formatURLConditional', function () {
+    test('We can format URL', function (assert) {
+      assert.strictEqual(
+        formatURLConditional('https://emberjs.com/logos#Branding'),
+        'https://emberjs.com/logos/#Branding',
+        'If # in URL, we add / before #'
+      );
+      assert.strictEqual(
+        formatURLConditional('http://localhost:4200/mascots?filter=tomster/'),
+        'http://localhost:4200/mascots?filter=tomster',
+        'If ? in URL, we remove trailing /'
+      );
+      assert.strictEqual(
+        formatURLConditional('http://localhost:4200/community'),
+        'http://localhost:4200/community/',
+        'If neither # or ? in URL, we add trailing /'
+      );
+    });
```

Comment:
Thanks for providing realistic URLs as examples. I'd suggest that we create a nested module for each of the 3 cases:

```js
module('Unit | Utility | format-url', function () {
  module('When # is in the URL', function () {
    test('We add / before #', function (assert) {
      const url = formatURLConditional('https://emberjs.com/logos#Branding');

      assert.strictEqual(
        url,
        'https://emberjs.com/logos/#Branding',
        'If # in URL, we add / before #'
      );
    });

    test('formatURLConditional is idempotent', function (assert) {
      const url = formatURLConditional('https://emberjs.com/logos#Branding');

      assert.strictEqual(
        formatURLConditional(url),
        'https://emberjs.com/logos/#Branding',
        'We get the same URL back.'
      );
    });
  });

  module('When ? is in the URL', function () {
    ...
  });

  module('When neither # nor ? is in the URL', function () {
    ...
  });
});
```

I don't think we need a nested module for `formatURLConditional` (as shown in line 5), since the utility currently exports only 1 function.

I suggest that we also test that `formatURLConditional` is idempotent. That is, when you run the function again, we get the same result:

```
f(f(x)) === f(x)
```